### DISCUSSION TITLE
How to use 'bzl' (starlark) LSP

### DISCUSSION THREAD

**Discussion by DanielKongsgaard** [mason-org/mason.nvim#1591] [Q&A]
`:Mason` has a language server called `bzl`, but when I try to ensure it is installed with `mason-lspconfig` I get:
```
[mason-lspconfig.nvim] Server "bzl" is not a valid entry in ensure_installed. Make sure to only provide lspconfig server names.
```

Also, I tried to set it up without `mason-lspconfig`, but had no luck so far.

How am I supposed to setup this LSP?

**Comment by DanielKongsgaard** [comment]
I figured out that I can set it up so it works with the following in `.config/nvim/after/ftplugin/bzl.lua`:
```lua
-- setup bazel/starlark lsp
if vim.fn.executable('bzl') == 1 then
    vim.lsp.start({
        name = 'Bazel/Starlark Language Server',
        cmd = { 'bzl', 'lsp', 'serve' },
    })
end
```
but I am not sure how to best give root dir.

If anyone has any experience with setting this LSP up I would still like some feedback though.

**Comment by tienshuoc** [comment]
Way late to the thread, but using `starpls` (https://mason-registry.dev/registry/list?search=bazel#starpls) directly for me works. You can just do:
```lua
:MasonInstall starpls
```
or
```
    mason_lspconfig.setup({
    ...
      ensure_installed = {
        "starpls", -- Bazel
      },
     ...
    })
```

**Reply by DanielKongsgaard** [reply]
I also ended up switching to `starpls` since it worked better last I checked.
